### ISSUE TITLE
dev roadmap

### ISSUE THREAD

**Issue by hidal00p** [hidal00p/i-hate-drones#1]
Define the core milestones to achieve in terms of program components and routines?

## Define software stack
Composing a software stack is the most important part of this work, since my code will only be a user friendly wrapper around all the horsepower provided by NN engine, Gymnasium env, RL algorithms, hyperparam tuning.

- pybullet
- PID or betaflight interface?
- torch? rllib? optuna?

**Concerns**
Is it a feasible solution to train an RL agent using betaflight as a flight controller. Or is it an overhead?

## Project workflow and user interaction mechanics
- How will user interface with it?
- How will the program interface with the available machine/cluster?
- What metric will be stored as a progress log?
- How will the training environment parameters be captured and stored? Do they contribute to reproducibility?

## Which problem do I want to tackle?
- Do I actually want to work with lidar, or maybe some path following is sufficient?

**Comment by ankovalov** [comment]
This issue would be considered solved, once a suitable README exists, which addresses most of the bullet points.

**Comment by ankovalov** [comment]
This becomes a critical point again, since I need to think how do I formulate my RL problem?

**Comment by ankovalov** [comment]
I decided to stick with the obstacle avoidance problem. This is probably the simplest approach, which will make use of the CtrlAviary environment. 

Based on the this I am going to formulate the next work pkgs.
